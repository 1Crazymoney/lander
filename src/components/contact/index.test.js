import React from 'react';
import Contact, {ContactDialogContent} from './index';
import renderer from 'react-test-renderer';


it('1- ContactDialogContent - render', () => {
  const props = {
    contact: {
      email: 'foo@bar.com',
      phone: '+90 501 223 34 43',
      website: 'http://www.zoogle.com',
      address: 'foo bar baz street 12 NY'
    }
  };

  const component = <ContactDialogContent {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('2- Contact - no contact data. should render null. (should run without contact data)', () => {
  const props = {

  };

  const component = <Contact {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('3- Contact - in edit mode. some data passed', () => {
  const props = {
    editMode: true,
    contact: {
      email: 'foo@bar.com'
    }
  };

  const component = <Contact {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('4- Contact - not in edit mode. some data passed', () => {
  const props = {
    editMode: false,
    contact: {
      email: 'foo@bar.com'
    }
  };

  const component = <Contact {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

