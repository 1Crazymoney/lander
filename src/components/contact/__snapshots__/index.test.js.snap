// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- ContactDialogContent - render 1`] = `
<div
  className="contact-dialog-body"
>
  <div
    className="contact-section"
  >
    <div
      className="section-icon"
    >
      <svg
        data-icon="envelope"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
          fill="currentColor"
        />
      </svg>
    </div>
    <a
      className="section-label"
      href="mailto:foo@bar.com"
    >
      foo@bar.com
    </a>
  </div>
  <div
    className="contact-section"
  >
    <div
      className="section-icon"
    >
      <svg
        data-icon="phone-alt"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M497.39 361.8l-112-48a24 24 0 0 0-28 6.9l-49.6 60.6A370.66 370.66 0 0 1 130.6 204.11l60.6-49.6a23.94 23.94 0 0 0 6.9-28l-48-112A24.16 24.16 0 0 0 122.6.61l-104 24A24 24 0 0 0 0 48c0 256.5 207.9 464 464 464a24 24 0 0 0 23.4-18.6l24-104a24.29 24.29 0 0 0-14.01-27.6z"
          fill="currentColor"
        />
      </svg>
    </div>
    <a
      className="section-label"
      href="tel:+905012233443"
    >
      +90 501 223 34 43
    </a>
  </div>
  <div
    className="contact-section"
  >
    <div
      className="section-icon"
    >
      <svg
        data-icon="globe"
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
          fill="currentColor"
        />
      </svg>
    </div>
    <a
      className="section-label"
      href="http://www.zoogle.com"
    >
      http://www.zoogle.com
    </a>
  </div>
  <div
    className="contact-section"
  >
    <div
      className="section-icon"
    >
      <svg
        data-icon="map-marker-alt"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="section-label"
    >
      foo bar baz street 12 NY
    </div>
  </div>
</div>
`;

exports[`2- Contact - no contact data. should render null. (should run without contact data) 1`] = `null`;

exports[`3- Contact - in edit mode. some data passed 1`] = `
<div
  className="contact edit-mode"
>
  <div
    className="edit-btn"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-pen fa-w-16"
      data-icon="pen"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
        fill="currentColor"
      />
    </svg>
  </div>
  <a
    className="contact-button"
  >
    Contact Me
  </a>
</div>
`;

exports[`4- Contact - not in edit mode. some data passed 1`] = `
<div
  className="contact"
>
  <a
    className="contact-button"
    href="#contact"
    onClick={[Function]}
  >
    Contact Me
  </a>
</div>
`;
